data/vibe/data/bson.d

```diff
@@ -395,7 +395,17 @@ struct Bson {
 	/// ditto
 	@property T get(T)()
 	const {
-		static if( is(T == double) ){ checkType(Type.double_); return fromBsonData!double(m_data); }
+		static if( is(T == double) ){
+			checkType(Type.double_, Type.long_, Type.int_);
+			if (m_type == Type.double_)
+				return fromBsonData!double(m_data);
+			else if (m_type == Type.int_)
+				return cast(double)fromBsonData!int(m_data);
+			else if (m_type == Type.long_)
+				return cast(double)fromBsonData!long(m_data);
+			else
+				assert(false);
+		}
```

Comment:
This contradicts the behavior specified in the method documentation. `opt` already converts to the requested type if possible. We might also add a `to` method, like `Json.to`.

Reply:
I was thinking if it should be put into a separate method, which would of course be better for backwards compatibility. I have currently implemented it through `get` because `get!string` also already covered multiple possible database types which are representable as strings, just like UUID also works on the same data type as ubyte[].

If I make a separate `to` function, what else should we support in conversion?